### PR TITLE
feat: send pdf attachemnt with image

### PR DESCRIPTION
**Pull request by rephus**
Closes: partially closes 6678 (Requires frontend)

### Description:

- [ ] If a user selects `image` as the export option, there is a toggle below the format which allows you to `Also include image as PDF attachment in email`
image

- [ ] by default, this toggle is `off` 
- [x] This is only for email to start with (we can open a separate issue for Slack)
- [x] add analytics

Screenshot from 2023-09-05 15-54-33
Screenshot from 2023-09-05 15-54-47

Sample pdf 
[scale-to-fit.pdf](https://github.com/lightdash/lightdash/files/12524497/scale-to-fit.pdf)

### CODE REVIEW

**Comment by rephus** [comment]
> is this a draft cause you think we should add the UI bits to this review

Good question, I was planning to do the frontend here as well, but now you mention,w e could merge this and implemente the frontend separately... what's your preference ?

File: packages/backend/src/services/UnfurlService/UnfurlService.ts [hunk 1]
```diff
@@ -187,12 +189,35 @@ export class UnfurlService {
...
     }
 
+    static async createImagePdf(
+        imageId: string,
+        buffer: Buffer,
+    ): Promise<string> {
+        const pdfDoc = await PDFDocument.create();
+        const pngImage = await pdfDoc.embedPng(buffer);
+        const page = pdfDoc.addPage();
+        const { width, height } = page.getSize();
```
[the remark this answers is not in the document]
Comment (rephus) [reply packages/backend/src/services/UnfurlService/UnfurlService.ts:199]: by default it should be an A4 pdf and 300ppp, we could set a custom page size, but I guess we could keep this default and wait for feedback.

File: packages/backend/src/services/UnfurlService/UnfurlService.ts [hunk 2]
```diff
@@ -187,12 +189,35 @@ export class UnfurlService {
...
+        const scaled = pngImage.scaleToFit(width, height);
+        page.drawImage(pngImage, {
+            x: 0,
+            y: 0,
+            width: scaled.width,
+            height: scaled.height,
+        });
+        const path = `/tmp/${imageId}.pdf`;
+        const pdfBytes = await pdfDoc.save();
```
[the remark this answers is not in the document]
Comment (rephus) [reply packages/backend/src/services/UnfurlService/UnfurlService.ts:208]: Good question. I'm not worried about the limit on the server, but google for example  has 25mb  limit on their attachements. Currently existing images are not added as attachements, but uploaded to S3 and downloaded via URL. 

My jaffle shop PDF was ~150kb, so we should be ok for now... (attaching CSVs could be problematic though)

**Comment by rephus** [comment]
Added `withPdf:boolean` to `scheduler_notification_job.completed` analytic event @TuringLovesDeathMetal

**Comment by TuringLovesDeathMetal** [comment]
niiice! Can we add it to `created` + `started` as well please 🙏 thanks so much @rephus !

**Comment by rephus** [comment]
> niiice! Can we add it to created + started as well please 🙏 thanks so much @rephus !

No, we can't, sorry, because  `created + started ` is triggered before we fetch the data about the specific scheduler, so we don't know if the scheduler was requested with pdf or not

**Comment by hamzahc1** [comment]
Looking at the original issue Acceptance Criteria for this - in the UI, is the intention that you select `Image` and then `PDF` or is PDF meant to be it's own format?

**Comment by rephus** [comment]
> Looking at the original issue Acceptance Criteria for this - in the UI, is the intention that you select `Image` and then `PDF` or is PDF meant to be it's own format?

You select `image` and then there is a toggle to send the image in pdf format as wella s the image in the email.

**Comment by rephus** [comment]
Frontend will be done in a separate PR

**Pull request merged**